### DISCUSSION TITLE
Immutable capsules usage

### DISCUSSION THREAD

**Discussion by busslina** [GregoryConrad/rearch-dart#44] [Q&A] [answered]
Is okay for a capsule that I know will not change, to use `Container.read` instead of `use`.

Is this way more efficient because it doesn't register any logic for that capsule?

**Comment by GregoryConrad** [comment, marked as answer]
Where would you call this container.read? If the answer is from within a capsule or RearchConsumer (which will be 99.99% of the time unless you’re making a web server), then no, do not use container.read. 

The overhead of `use(someCapsule)` is negligible and when dependencies are cached it is O(1) (at least in the dart implementation, in Rust it’s O(logn) iirc because of the concurrency model).

In capsules/RearchConsumer, always `use(someCapsule)`. It will ensure you don’t shoot yourself in the foot when things change in the future.

**Reply by busslina** [reply]
Yeah, it was inside another capsule.

I will do as you say
